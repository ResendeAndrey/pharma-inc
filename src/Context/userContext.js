import { createContext, useContext, useState} from 'react';


export const UserContext = createContext();

export function UserContextProvider({children}) {
  const localStorageUsers = JSON.parse(localStorage.getItem("users"));
  const [users, setUsers] = useState(localStorageUsers === null ? [] : localStorageUsers);
  const [openModal, setOpenModal] = useState(false);


 function saveUsers(data) {
    setUsers([...new Set([...users, ...data])])
  }

  return ( 
    <UserContext.Provider
      value={{
        users,
        saveUsers,
        openModal,
        setOpenModal
      }}
    >
      {children}
    </UserContext.Provider>
  )
}

export const useUser = () => {
  return useContext(UserContext);
}