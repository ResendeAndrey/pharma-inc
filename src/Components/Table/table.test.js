import { shallow } from 'enzyme'
import React from 'react'
import { UserContextProvider } from '../../Context/userContext' 
import TableComponent from './table'

const sut = shallow(
<UserContextProvider>
  <TableComponent userItems={[]}/>
</UserContextProvider>
);

describe(('Table Component'), () => {
  afterEach(() => jest.clearAllMocks())
  it('Should match snapshot', () => {
    expect(sut.html()).toMatchSnapshot();
  })
  it('Should return data null', () => {
    expect(sut.find('td')).toEqual({});
  })
  it('Should return data', () => {
    const users = {
      "results": [
        {
          "gender": "female",
          "name": {
            "title": "Ms",
            "first": "Pamela",
            "last": "Fisher"
          },
          "dob": {
            "date": "1973-08-27T01:12:51.947Z",
            "age": 48
          },
          "registered": {
            "date": "2009-03-27T20:36:19.038Z",
            "age": 12
          },
          "nat": "US"
        }
      ],
      "info": {
        "seed": "6dea3e005ec27ce9",
        "results": 1,
        "page": 1,
        "version": "1.3"
      }
    }
    const data = shallow(
      <UserContextProvider>
        <TableComponent userItems={users}/>
      </UserContextProvider>
      );

    expect(data.find('td')).toEqual({})
  })
})