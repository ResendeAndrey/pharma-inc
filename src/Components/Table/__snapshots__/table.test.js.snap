// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Component Should match snapshot 1`] = `"<div class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root\\"><table class=\\"MuiTable-root table css-rqglhn-MuiTable-root\\" aria-label=\\"caption table\\"><thead class=\\"MuiTableHead-root css-15wwp11-MuiTableHead-root\\"><tr class=\\"MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root\\"><th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root\\" scope=\\"col\\">Name</th><th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root\\" scope=\\"col\\">Gender</th><th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root\\" scope=\\"col\\">Birth</th><th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root\\" scope=\\"col\\">Actions</th></tr></thead><tbody class=\\"MuiTableBody-root css-apqrd9-MuiTableBody-root\\"></tbody></table></div>"`;

exports[`Table Component shoud return data null 1`] = `"<div class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root\\"><table class=\\"MuiTable-root table css-rqglhn-MuiTable-root\\" aria-label=\\"caption table\\"><thead class=\\"MuiTableHead-root css-15wwp11-MuiTableHead-root\\"><tr class=\\"MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root\\"><th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root\\" scope=\\"col\\">Name</th><th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root\\" scope=\\"col\\">Gender</th><th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root\\" scope=\\"col\\">Birth</th><th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root\\" scope=\\"col\\">Actions</th></tr></thead><tbody class=\\"MuiTableBody-root css-apqrd9-MuiTableBody-root\\"></tbody></table></div>"`;
